subsys/bluetooth/host/gatt.c

```diff
@@ -2130,23 +2138,37 @@ static int gatt_notify_mult(struct bt_conn *conn, uint16_t handle,
 	}
 
 	if (!*buf) {
-		*buf = bt_att_create_pdu(conn, BT_ATT_OP_NOTIFY_MULT,
-					 sizeof(*nfy) + params->len);
+		*buf = bt_att_create_pdu(conn, BT_ATT_OP_NOTIFY_MULT, pdu_len);
 		if (!*buf) {
 			return -ENOMEM;
 		}
 
-		bt_att_set_tx_meta_data(*buf, params->func, params->user_data);
+		bt_att_set_tx_meta_data(*buf, tuple[0].func, tuple[0].user_data);
 	}
 
-	BT_DBG("handle 0x%04x len %u", handle, params->len);
+	for (i = 0; i < num_params; i++) {
+		/* Check if attribute is a characteristic then adjust the handle */
+		if (!bt_uuid_cmp(tuple[i].attr->uuid, BT_UUID_GATT_CHRC)) {
+			struct bt_gatt_chrc *chrc = tuple[i].attr->user_data;
 
-	nfy = net_buf_add(*buf, sizeof(*nfy));
-	nfy->handle = sys_cpu_to_le16(handle);
-	nfy->len = sys_cpu_to_le16(params->len);
+			if (!(chrc->properties & BT_GATT_CHRC_NOTIFY)) {
+				return -EINVAL;
+			}
 
-	net_buf_add(*buf, params->len);
-	memcpy(nfy->value, params->data, params->len);
+			handle = bt_gatt_attr_value_handle(tuple[i].attr);
+		} else {
+			handle = bt_gatt_attr_get_handle(tuple[i].attr);
+		}
+
+		BT_DBG("handle 0x%04x len %u", handle, tuple[i].len);
+
+		nfy = net_buf_add(*buf, sizeof(*nfy));
+		nfy->handle = sys_cpu_to_le16(handle);
+		nfy->len = sys_cpu_to_le16(tuple[i].len);
+
+		net_buf_add(*buf, tuple[i].len);
+		memcpy(nfy->value, tuple[i].data, tuple[i].len);
```

Comment:
```suggestion
		(void)memcpy(nfy->value, tuple[i].data, tuple[i].len);
```

Reply:
what void is used for before memcpy?

Reply:
At this point is it just convention used by the Bluetooth subsystem. It is partially based on rule 17.4 from https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?highlight=coding%20guideline#main-rules